pkg/controller/workflow/delete_handler.go

```diff
@@ -0,0 +1,30 @@
+package workflow
+
+import (
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+type DeleteEventHandler struct {
```

Comment:
can we move this to `watch` package as it was designed for these kinds of primitives?